### DISCUSSION TITLE
Announcing Perseus v0.3.0!

### DISCUSSION THREAD

**Discussion by arctic-hen7** [framesurge/perseus#95] [Announcements]
I am very happy to announce that Perseus v0.3.0 has now gone stable! There's a guide [here](https://arctic-hen7.github.io/perseus/en-US/docs/0.3.x/updating) on how to upgrade from v0.2.0, but here are some of the most notable features:

- Added seamless deployment process
- Added support for hosting exported sites under relative paths
- Made creating templates _far_ more ergonomic (`Rc`s are out, automatic de/serialization of properties is in)
- Added a plugins system that enables extending Perseus in almost any way imaginable
- Added a plugin to automate size optimizations (see [here](https://github.com/arctic-hen7/perseus-size-opt))
- Made exports much more logical
- Feature-gated opt-in features to reduce Perseus binary sizes from around 1MB to less than 300KB (even less without i18n)
- Added `perseus snoop` to investigate internal logs
- Added a server integration for [Warp](https://github.com/seanmonstar/warp)
- Upgraded to [Sycamore](https://github.com/sycamore-rs/sycamore) v0.7.0 (which adds hydration!)
- Made locale redirection much faster, and fully functional even without JS or Wasm (meaning Perseus fully supports progressive enhancement)

And, last but not least, Perseus now has a fully functional [website](https://arctic-hen7.github.io/perseus/en-US) with Lighthouse scores of 100 on desktop and 96 on mobile!

This has been 26 beta versions in the making over the last three months, and I'd like to thank everyone who was involved, especially @lukechu10 for his astonishing work on Sycamore v0.7.0 and @phaleth for discovering a security vulnerability in the Actix Web integration (now patched, see [here](https://github.com/arctic-hen7/perseus/security/advisories/GHSA-gjrj-9rj4-pgwx) for the details).

Additionally, I'd like to personally thank the community for helping this project reach well over 300 stars, it's been truly amazing to have so much support!

**Comment by rieval** [comment]
Great work!

**Reply by arctic-hen7** [reply]
Thank you very much!
